dotCMS/src/main/java/com/dotcms/graphql/datafetcher/page/VisitorDataFetcher.java

```diff
@@ -0,0 +1,45 @@
+package com.dotcms.graphql.datafetcher.page;
+
+import com.dotcms.graphql.DotGraphQLContext;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.PermissionLevel;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.htmlpageasset.business.HTMLPageAssetAPI;
+import com.dotmarketing.portlets.htmlpageasset.business.render.page.JsonMapper;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.liferay.portal.model.User;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+import java.io.CharArrayReader;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * This DataFetcher returns a {@link Map} representing a {@link Template} associated to the originally
+ * requested {@link HTMLPageAsset}.
+ */
+public class VisitorDataFetcher implements DataFetcher<Map<Object, Object>> {
+    @Override
+    public Map<Object, Object> get(final DataFetchingEnvironment environment) throws Exception {
+        try {
+            final User user = ((DotGraphQLContext) environment.getContext()).getUser();
+            final Contentlet contentlet = environment.getSource();
+
+
+        } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6028986)